### PR TITLE
Fix validate button with variant

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx [hunk 1]
```diff
@@ -68,45 +64,36 @@ const SensitivityTable: FunctionComponent<SensitivityTableProps> = ({
...
+                        <TableCell
+                            key={column.dataKey}
+                            sx={{ width: column.width, textAlign: 'center' }}
+                        >
+                            <Box
+                                sx={{
+                                    justifyContent: 'space-between',
+                                    alignItems: 'center',
+                                    fontWeight: 'normal',
+                                    margin: 1,
```
Comment (ghelloumi) [inline src/components/dialogs/parameters/sensi/sensitivity-table.tsx:79]: - font-weight: design will not be the same as other components using Table Cell, I think  that you will not need to modify this.
- margin: it is not necessary

anyway, I don't think that this box is necessary, even if it is necessary, applying "justifyContent" and "alignItem" outside a flex context (display: flex) will not have an effect on style.

File: src/components/dialogs/parameters/sensi/columns-definitions.ts [hunk 2]
```diff
@@ -116,15 +114,14 @@ export const COLUMNS_DEFINITIONS_INJECTIONS_SET = [
...
-        width: '25%',
     },
     {
         label: 'Active',
         dataKey: ACTIVATED,
         initialValue: true,
         checkboxItems: true,
         editable: true,
-        width: '5%',
+        width: '5rem',
```
Comment (ghelloumi) [inline src/components/dialogs/parameters/sensi/columns-definitions.ts:124]: can you add'width' to IColumnsDef

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx [hunk 3]
```diff
@@ -68,45 +64,36 @@ const SensitivityTable: FunctionComponent<SensitivityTableProps> = ({
...
-                                    <FormattedMessage id={column.label} />
-                                </Box>
-                            </TableCell>
-                        ))}
-                        <TableCell>
-                            <Tooltip
-                                title={intl.formatMessage({
-                                    id: 'AddRows',
-                                })}
+                    {columnsDefinition.map((column: any) => (
```
Comment (ghelloumi) [inline src/components/dialogs/parameters/sensi/sensitivity-table.tsx:69]: column have the type  IColumnsDef (you can added also to SensitivityTableProps)

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx [hunk 4]
```diff
@@ -68,45 +64,36 @@ const SensitivityTable: FunctionComponent<SensitivityTableProps> = ({
...
+                            })}
+                        >
+                            <IconButton onClick={handleAddRowsButton}>
+                                <AddCircleIcon />
+                            </IconButton>
+                        </Tooltip>
+                    </TableCell>
                 </TableHead>
                 <TableBody>
                     {currentRows.map((row: any, index: number) => (
```
Comment (ghelloumi) [inline src/components/dialogs/parameters/sensi/sensitivity-table.tsx:99]: row have the type: UseFieldArrayReturn from 'react-hook-form/dist/types';

File: src/components/dialogs/parameters/sensi/table-cell.tsx [hunk 5]
```diff
@@ -37,8 +39,15 @@ function EditableTableCell(
                     name={`${arrayFormName}[${rowIndex}].${column.dataKey}`}
                     options={column.equipmentTypes}
                     disableClearable={true}
+                    allowNewValue
                     size="small"
-                    sx={{ width: getDynamicWidth(column.equipmentTypes) }}
+                    fullWidth
+                    sx={{
+                        flexDirection: 'row',
```
Comment (ghelloumi) [inline src/components/dialogs/parameters/sensi/table-cell.tsx:46]: you can delete this property, not a flex context

File: src/components/dialogs/parameters/sensi/table-row.tsx [hunk 6]
```diff
@@ -51,14 +49,10 @@ const TableRowComponent: FunctionComponent<TableRowComponentProps> = ({
                 >
                     <span hidden={!hover}>
                         <IconButton
-                            color="primary"
+                            sx={{ border: 'solid 0px rgba(0,0,0,0.26)' }}
```
Comment (ghelloumi) [inline src/components/dialogs/parameters/sensi/table-row.tsx:52]: maybe we can create a new field in global theme for the color of the border

**Review by ghelloumi — CHANGES_REQUESTED** [review]

File: src/components/dialogs/parameters/sensi/table-row.tsx (hunk 6 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/table-row.tsx]: done

File: src/components/dialogs/parameters/sensi/table-cell.tsx (hunk 5 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/table-cell.tsx]: done

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx (hunk 4 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensitivity-table.tsx]: done

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx (hunk 3 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensitivity-table.tsx]: done

File: src/components/dialogs/parameters/sensi/columns-definitions.ts (hunk 2 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/columns-definitions.ts]: done

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx (hunk 1 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensitivity-table.tsx]: done

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx [hunk 7]
```diff
@@ -68,57 +65,40 @@ const SensitivityTable: FunctionComponent<SensitivityTableProps> = ({
...
-                    </TableRow>
+                    ))}
+                    <TableCell sx={{ width: '5rem', textAlign: 'center' }}>
+                        <Tooltip
+                            title={intl.formatMessage({
+                                id: 'AddRows',
+                            })}
+                        >
+                            <IconButton onClick={handleAddRowsButton}>
+                                <AddCircleIcon />
```
Comment (dbraquart) [inline src/components/dialogs/parameters/sensi/sensitivity-table.tsx:85]: just curious: why did this button become black & white, and not blue like in other forms, with the default theme ?

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

File: src/components/dialogs/parameters/sensi/sensitivity-table.tsx (hunk 7 above)
Comment (ghazwarhili) [reply src/components/dialogs/parameters/sensi/sensitivity-table.tsx:85]: requested by PO

**Review by ghelloumi — APPROVED** [review]

**Pull request merged**